rule-audit-client/src/components/capiFeed/CapiSearchOptions.tsx

```diff
@@ -2,97 +2,110 @@ import * as React from "react";
 import { connect } from "react-redux";
 import { Dispatch, bindActionCreators } from "redux";
 import { useEffect, useState } from "react";
-import Select from "react-select";
+import {
+  InputSupper,
+  QueryElement,
+  Filter,
+  SelectAsyncFilter,
+} from "@guardian/threads";
+import { debounce } from 'lodash';
 
 import AppTypes from "AppTypes";
 import { actions, selectors, thunks } from "redux/modules/capiContent";
 import {
-  CapiTag,
-  CapiSection,
   fetchCapiTags,
-  fetchCapiSections
+  fetchCapiSections,
 } from "services/capi";
 
 type IProps = ReturnType<typeof mapDispatchToProps> &
   ReturnType<typeof mapStateToProps>;
 
-type Selection = { value: string; label: string };
+const filters: (Filter | SelectAsyncFilter)[] = [
+  {
+    name: "Tag",
+    type: "select_async",
+    onInputChange: async (input: string) => {
+      const tags = await fetchCapiTags(input);
+      return (
+        tags?.results?.map((tag) => ({
+          label: tag.webTitle,
+          value: tag.id,
+        })) || []
+      );
+    },
+  },
+  {
+    name: "Section",
+    type: "select_async",
+    onInputChange: async (input: string) => {
+      const tags = await fetchCapiSections(input);
+      return (
+        tags?.results?.map((tag) => ({
+          label: tag.webTitle,
+          value: tag.id,
+        })) || []
+      );
+    },
+  },
+];
+
+const getSearchEntitiesFromQueryElements = (elements: QueryElement[]) => ({
+  query: elements
+    .filter((element) => element.type === "text")
+    .reduce((acc, el) => `${acc} ${el.value}`, ""),
```

Comment:
why the empty string here?

Reply:
As I understand it the accumulator starts with an empty string – otherwise we'll get a string 'undefined' as the value `undefined` is coerced.

Bit lazy as we'll always have two leading spaces at the start of this query, which I hadn't spotted. In fact, I'd completely forgotten about `.join`, which would be better! – I'll use that.